src/Hl7.Fhir.ElementModel.Tests/TypedElementOnSourceNodeTests.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Linq;
+using Hl7.Fhir.Serialization;
+using Hl7.Fhir.Specification;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Hl7.Fhir.ElementModel.Tests
+{
+    [TestClass]
+    public class TypedElementOnSourceNodeTests
+    {
+        private readonly IStructureDefinitionSummaryProvider provider = new PocoStructureDefinitionSummaryProvider();
+
+        [ExpectedException(typeof(StructuralTypeException), "Should have thrown on .Value as complex types can't have a value")]
```

Comment:
No! I never introduced this as an exception - it's not an exceptional circumstance, and I never coded any such constraints into the element model. There might be actually code checking .Value, irrespective of whether it is complex or not. It's not wrong to do that.

Reply:
There is a call to tryMapFhirPrimitiveTypeToSystemType in the Value Getter. If it's not a primitive raiseTypeError is called. This test was added to make sure that we don't break this part of the code.